device/core/test/_device_client_test.js

```diff
@@ -0,0 +1,257 @@
+  var sharedKeyConnectionString = 'HostName=host;DeviceId=id;SharedAccessKey=key';
+  var sharedAccessSignature = '"SharedAccessSignature sr=hubName.azure-devices.net/devices/deviceId&sig=s1gn4tur3&se=1454204843"';
+
+  describe('#fromConnectionString', function () {
+    /*Tests_SRS_NODE_DEVICE_CLIENT_05_006: [The fromConnectionString method shall return a new instance of the InternalClient object, as by a call to new InternalClient(new Transport(...)).]*/
+    it('returns an instance of Client', function () {
+      var client = Client.fromConnectionString(sharedKeyConnectionString, FakeTransport);
+      assert.instanceOf(client, Client);
+    });
+
+    it('doesn\'t try to renew the SAS token when using x509', function (testCallback) {
+      this.clock = sinon.useFakeTimers();
+      var clock = this.clock;
+
+      var x509ConnectionString = 'HostName=host;DeviceId=id;x509=true';
+      var client = new Client.fromConnectionString(x509ConnectionString, FakeTransport);
+      assert.instanceOf(client, Client);
+
+      sinon.stub(client._transport, 'updateSharedAccessSignature').callsFake(function () {
+        clock.restore();
+        testCallback(new Error('updateSharedAccessSignature should not have been called'));
+      });
+
+      this.clock.tick(3600000); // 1 hour: this should trigger the call to renew the SAS token.
+      clock.restore();
+      testCallback();
+    });
+
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_093: [The `fromConnectionString` method shall create a new `X509AuthorizationProvider` object with the connection string passed as argument if it contains an X509 parameter and pass this object to the transport constructor.]*/
+    it('creates a X509AuthenticationProvider and passes it to the transport', function (testCallback) {
+      var x509ConnectionString = 'HostName=host;DeviceId=id;x509=true';
+      Client.fromConnectionString(x509ConnectionString, function (authProvider) {
+        assert.instanceOf(authProvider, X509AuthenticationProvider);
+        testCallback();
+      });
+    });
+  });
+
+  describe('#fromSharedAccessSignature', function () {
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_030: [The fromSharedAccessSignature method shall return a new instance of the InternalClient object] */
+    it('returns an instance of Client', function () {
+      var client = Client.fromSharedAccessSignature(sharedAccessSignature, FakeTransport);
+      assert.instanceOf(client, Client);
+    });
+  });
+
+  describe('#fromAuthenticationProvider', function () {
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_091: [The `fromAuthenticationProvider` method shall return a `InternalClient` object configured with a new instance of a transport created using the `transportCtor` argument.]*/
+    it('returns an instance of Client', function () {
+      var client = Client.fromAuthenticationProvider({}, FakeTransport);
+      assert.instanceOf(client, Client);
+    });
+  });
+
+
+  describe('#uploadToBlob', function() {
```

Comment:
>uploadToBlob [](start = 13, length = 12)

random comment -- I haven't looked at the code yet, but does the blob upload functionality also need retry tests?